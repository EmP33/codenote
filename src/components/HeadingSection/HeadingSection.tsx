import React from "react";
import { Typography, Grid, Container } from "@mui/material";
import styled from "styled-components";
import { Link } from "react-router-dom";

const HeadingWrapper = styled.div`
  display: grid;
  justify-items: flex-start;
  grid-row-gap: 1rem;
`;
const RegisterButton = styled.button`
  position: relative;
  padding: 0.75rem 1.5rem 0.75rem 1.5rem;
  background: transparent;
  color: #fff;
  outline: none;
  cursor: pointer;
  font-size: 16px;
  font-weight: bold;
  letter-spacing: 1px;
  border: 1px solid #fff;
  transition: all 0.3s ease;
  z-index: 1;

  &:hover:after {
    top: 0;
    left: 0;
  }

  &:after {
    content: "";
    z-index: -1;
    position: absolute;
    width: 100%;
    height: 100%;
    top: 5px;
    left: 5px;
    transition: 0.2s;
    background-color: var(--color-tertiary-light);
  }
`;

const HeadingSection: React.FC = () => {
  return (
    <Container maxWidth="xl">
      <Grid
        container
        spacing={0}
        alignItems="center"
        sx={{
          height: "87.5vh",
          width: "100%",
        }}
      >
        <Grid item xs={12} sm={7} lg={5}>
          <HeadingWrapper>
            <Typography
              variant="h1"
              sx={{
                color: "var(--color-tertiary-light)",
                fontWeight: "bold",
                fontSize: { xs: 50, md: 100 },
              }}
            >
              CodeNote
            </Typography>
            <Typography variant="body1" sx={{ fontSize: { xs: 14, md: 18 } }}>
              Join us to increase your notes to a higher level. Lots of good
              features. Create your account and get started!
            </Typography>
            <Link to="/register">
              <RegisterButton>Create an Account</RegisterButton>
            </Link>
          </HeadingWrapper>
        </Grid>
      </Grid>
    </Container>
  );
};

export default HeadingSection;
