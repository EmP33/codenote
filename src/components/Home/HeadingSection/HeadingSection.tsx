import React from "react";
import { Typography, Grid, Container } from "@mui/material";
import styled from "styled-components";
import { Link } from "react-router-dom";
//@ts-ignore
import mockup from "../../../assets/mockup.png";

const HeadingWrapper = styled.div`
  display: grid;
  justify-items: flex-start;
  grid-row-gap: 1rem;
`;
const RegisterButton = styled.button`
  position: relative;
  padding: 0.75rem 1.5rem 0.75rem 1.5rem;
  background: transparent;
  color: #fff;
  outline: none;
  cursor: pointer;
  font-size: 16px;
  font-weight: bold;
  letter-spacing: 1px;
  border: 2px solid var(--color-tertiary-dark);
  transition: all 0.3s ease;
  z-index: 1;

  &:hover:after {
    top: 0;
    left: 0;
  }

  &:after {
    content: "";
    z-index: -1;
    position: absolute;
    width: 100%;
    height: 100%;
    top: 5px;
    left: 5px;
    transition: 0.2s;
    background-color: var(--color-tertiary-dark);
    @media screen {
    }
  }
`;

const HeadingSection: React.FC = () => {
  return (
    <Container maxWidth="xl">
      <Grid
        container
        spacing={0}
        alignItems="center"
        sx={{
          height: "87.5vh",
          width: "100%",
          mt: { xs: 2, lg: 0 },
        }}
      >
        <Grid item xs={12} lg={5}>
          <HeadingWrapper>
            <Typography
              variant="h1"
              sx={{
                color: "var(--color-tertiary-dark)",
                fontWeight: "bold",
                fontSize: { xs: 50, md: 100 },
              }}
            >
              CodeNote
            </Typography>
            <Typography
              variant="body1"
              sx={{
                fontSize: { xs: 14, md: 18 },
                color: "var(--color-tertiary-dark)",
              }}
            >
              Join us to increase your notes to a higher level. Lots of good
              features. Create your account and get started!
            </Typography>
            <Link to="/register">
              <RegisterButton>Create an Account</RegisterButton>
            </Link>
          </HeadingWrapper>
        </Grid>
        <Grid
          item
          xs={12}
          lg={7}
          sx={{
            display: "flex",
            justifyContent: "center",
            height: { xs: "60%", md: "70%", lg: "100%" },
          }}
        >
          <img
            src={mockup}
            alt="mockup"
            style={{
              width: "100%",
              height: "100%",
              objectFit: "contain",
            }}
          />
        </Grid>
      </Grid>
    </Container>
  );
};

export default HeadingSection;
