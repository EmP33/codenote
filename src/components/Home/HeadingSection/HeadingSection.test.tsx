import { shallow } from "enzyme";
import HeadingSection from "./HeadingSection";

it("should render HeadingSection component", () => {
  expect(shallow(<HeadingSection />)).toMatchSnapshot();
});
