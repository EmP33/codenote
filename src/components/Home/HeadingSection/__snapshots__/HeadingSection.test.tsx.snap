// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render HeadingSection component 1`] = `
<ForwardRef(Container)
  maxWidth="xl"
>
  <ForwardRef(Grid)
    alignItems="center"
    container={true}
    spacing={0}
    sx={
      Object {
        "height": "87.5vh",
        "width": "100%",
      }
    }
  >
    <ForwardRef(Grid)
      item={true}
      lg={5}
      sm={7}
      xs={12}
    >
      <styled.div>
        <ForwardRef(Typography)
          sx={
            Object {
              "color": "var(--color-tertiary-light)",
              "fontSize": Object {
                "md": 100,
                "xs": 50,
              },
              "fontWeight": "bold",
            }
          }
          variant="h1"
        >
          CodeNote
        </ForwardRef(Typography)>
        <ForwardRef(Typography)
          sx={
            Object {
              "fontSize": Object {
                "md": 18,
                "xs": 14,
              },
            }
          }
          variant="body1"
        >
          Join us to increase your notes to a higher level. Lots of good features. Create your account and get started!
        </ForwardRef(Typography)>
        <Link
          to="/register"
        >
          <styled.button>
            Create an Account
          </styled.button>
        </Link>
      </styled.div>
    </ForwardRef(Grid)>
  </ForwardRef(Grid)>
</ForwardRef(Container)>
`;
